[Bernard Wan De Yuan] iP

bernardwan opened the pull request:
### DUKE 

> “Your mind is for having ideas, not holding them.” – David Allen (source)

Hi this is my Duke program! It's

- _Fun_ 👍 
- Easy to use
- text-based

Steps to use:

1. Download from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/461)
2. Run
3. **Use!**

Features:

- [x] Managing tasks
- [ ] GUI

Sample code `main` :
```java
public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

bernardwan committed:
Level-1

bernardwan committed:
Level-2

bernardwan committed:
Level-3

bernardwan committed:
Level-4

bernardwan committed:
Added automated text UI testing

bernardwan committed:
Implemented Level-5, updated test cases

bernardwan committed:
Implemented Level-6. updated tests

bernardwan committed:
Implemented Level-7

bernardwan committed:
Implemented Level-8

bernardwan committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Task.java
#	text-ui-test/EXPECTED.TXT

bernardwan committed:
Updated tests

bernardwan committed:
Resolved conflict errors

bernardwan committed:
Refactor code to make it more OOP

bernardwan committed:
Organise into package

bernardwan committed:
Add Junit tests

bernardwan committed:
Create JAR File

bernardwan committed:
Add Javadocs

bernardwan committed:
Tweak code to follow coding standard

bernardwan committed:
Implement Level-9 Find feature

bernardwan committed:
Merge branch 'branch-A-CodingStandard'

bernardwan committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/TaskList.java

bernardwan committed:
Fix minor javadocs errors

david-eom commented on src/test/java/dukeTest/TaskListTest.java:
```diff
@@ -0,0 +1,31 @@
+package dukeTest;
+
+import duke.DukeException;
+import duke.Task;
+import duke.TaskList;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
```
Should `java` related imports come before `duke`?

david-eom commented on src/main/java/duke/Command/AddCommand.java:
```diff
@@ -0,0 +1,42 @@
+package duke.Command;
+
+import duke.*;
```
I think perhaps can consider importing individual classes instead of using wildcard `import`!

david-eom commented on src/main/java/duke/Command/AddCommand.java:
```diff
@@ -0,0 +1,42 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task task;
+        if(type.equals("todo")) {
+            task = new Task.Todo(description, false);
+
+        } else if(type.equals("deadline")) {
+            String[] temp = description.split("by ", 2);
+            task = new Task.Deadline(temp[0], false, temp[1]);
+
+        } else if(type.equals("event")) {
+            String[] temp = description.split("at ", 2);
+            task = new Task.Event(temp[0], false, temp[1]);
```
Maybe can consider using `enum` for this!

david-eom commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+     * Loads task data from storage
+     *
+     * @return ArrayList of tasks
+     * @throws DukeException
```
Feel like the condition when the `DukeException` is thrown should be specified here :))

david-eom commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,161 @@
+        this.status= status;
+    }
+
+    public static class Todo extends Task{
```
Should there be a whitespace before the curly braces?

david-eom commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,58 @@
+        String first_word = input.split(" ")[0];
+
+
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (first_word.equals("done")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.split(" ")[1]) - 1;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter an integer after 'done'. (e.g. done 2)");
+            }
+            return new DoneCommand(index);
+        } else if (first_word.equals("delete")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.split(" ")[1]) - 1;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter an integer after 'delete'. (e.g. delete 2)");
+            }
+            return new DeleteCommand(index);
+        } else if (first_word.equals("find")){
+            String remaining;
+            try {
+                remaining = input.split(" ", 2)[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter a keyword after 'find'.");
+            }
+            return new FindCommand(remaining);
+        } else {
+            String remaining;
+            try {
+                remaining = input.split(" ", 2)[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, tasks must include descriptions.");
+            }
+            return new AddCommand(first_word, remaining);
+        }
```
Perhaps we can use `enum` here to simplify the logic here! Also, personally I fell that all the `try-catch` statement here can be abstracted to other class method e.g. `createDone()` etc. to prevent arrowhead style code, and that one single `catch` for `ArrayIndexOutOfBoundsException` could be used for the entire method. Don't know if I articulate my idea in a very accurate way 😂

david-eom commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+            String line = reader.nextLine();
+            String[] data = line.split(" \\| ");
+            switch (data[0]) {
+                case "todo":
```
Should `case` be on the same indent level as `switch`?

david-eom reviewed (commented):
LGTM! The code looks clean and easy to read, with most of the comments being nit-picky 😂 Think that the logic can be a bit more simplified in `Parser`, and that some javadoc is missing. Excellent code, keep up the good work!! 👍

bernardwan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chengseong commented on src/main/java/duke/Command/FindCommand.java:
```diff
@@ -0,0 +1,30 @@
+package duke.Command;
+
+import duke.*;
```
I think you should import package by package, for example `import duke.TaskList` rather than using `import duke.*`

chengseong commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,5 @@
+public abstract class Command {
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    public abstract boolean isExit();
```
You might want to consider setting `isExit` to `False` in the abstract, then only changing it to `True` when it is an ExitCommand. This way you do not have to constantly check what Command it is

chengseong reviewed (commented):
LGTM! I think the code is very clean and easy to read and there is very clear abstraction between classes. Well done!

bernardwan committed:
Add Gradle to project

bernardwan committed:
Add ability to use checkstyle

bernardwan committed:
Add GUI

bernardwan committed:
Modify config for proper Gradle usage

Gradle run and shadowJar tasks are not functioning as intended.

This should be fixed for ease of use and to create a new JAR release.

Let's edit the build.gradle for proper main class declaration.
Also,let's fix any checkstyle errors that might inhibit the build process.

bernardwan committed:
Add Assert statements

bernardwan committed:
Check code for code quality and clean up code

There are some left over code from before GUI was implemented.

These leftover code that is not used should be removed to improve
code quality and readability.

Let's check through the code and remove any unused methods or
other files.

bernardwan committed:
Merge pull request #1 from bernardwan/branch-A-Assertions

branch-A-Assertions

bernardwan committed:
Merge branch 'master' into branch-A-CodeQuality

bernardwan committed:
Merge pull request #2 from bernardwan/branch-A-CodeQuality

Branch a code quality

bernardwan committed:
Implement B-DoWithinPeriodTasks

bernardwan committed:
Refactor code

bernardwan committed:
Add User Guide

bernardwan committed:
Rename boolean variable

bernardwan committed:
Improve GUI

bernardwan committed:
Handle some exceptions and change font

bernardwan committed:
Update README.md
